### PR TITLE
Inspec Hiera 3 and Upgraded Global to Heira 5

### PR DESCRIPTION
**Pull request by phillipi-shinesolutions**

### CODE REVIEW

File: test/inspec/spec_helper.rb [hunk 1]
```diff
@@ -8,5 +8,6 @@ def init_conf
     '::packer_build_name' => Facter.value(:packer_build_name),
     '::os' => Facter.value(:os)
   }
-  @hiera = Hiera.new(config: Facter.value(:packer_staging_dir) + '/hiera.yaml')
+  # TODO: Fix this when inspec has a path from hiera 3 to 5
```
Comment (cliffano) [inline test/inspec/spec_helper.rb]: Please add a link to the discussion thread that explains why hiera gem is deprecated, that thread also has a good pointer to look at rspec-puppet.

File: conf/puppet/hieradata/common.yaml [hunk 2]
```diff
@@ -18,6 +18,7 @@ aem_curator::install_author::repository_volume_device: /dev/xvdb
 aem_curator::install_author::repository_volume_mount_point: /mnt/ebs1
 aem_curator::install_author::post_stop_sleep_secs: 300
 aem_curator::install_author::aem_license_base: "%{lookup('tmp_dir')}/license"
+aem_curator::install_author::aem_keystore_password: "%{lookup('AEM_Keystore_Password')}"
```
Comment (cliffano) [inline conf/puppet/hieradata/common.yaml]: This isn't related to hiera upgrade, please move to a separate feature branch / PR.

File: Puppetfile [hunk 3]
```diff
@@ -14,6 +14,7 @@ mod 'puppetlabs/concat', '4.1.0'
 mod 'puppetlabs/stdlib', '4.25.1'
 mod 'thias/sysctl', '1.0.6'
 mod 'MiamiOH/tcpwrappers', '1.1.0'
+mod 'luizvasconceloss-hiera_ssm_paramstore', '0.2.2'
```
Comment (cliffano) [inline Puppetfile]: This isn't related to hiera upgrade, please move to a separate feature branch / PR.

File: Gemfile [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 source 'https://rubygems.org'
 
+gem 'aws-sdk-core'
```
Comment (cliffano) [inline Gemfile]: This isn't related to hiera upgrade, please move to a separate feature branch / PR.

File: conf/puppet/hiera.yaml [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
+      - docker.yaml
+      - author-publish-dispatcher.yaml
+      - local.yaml
+      - "%{::os.family}/%{::os.name}-%{::os.release.major}.yaml"
+      - "%{::os.family}/%{::os.name}.yaml"
+      - "%{::platform_type}.yaml"
+      - "%{::component}.yaml"
+      - common.yaml
+
+  - name: "AWS Parameter Store"
```
Comment (cliffano) [inline conf/puppet/hiera.yaml]: This isn't related to hiera upgrade, please move to a separate feature branch / PR.

File: conf/puppet/hiera.yaml [hunk 6]
```diff
@@ -1,14 +1,26 @@
...
-:merge_behavior: native
-:deep_merge_options: {}
+version: 5
+defaults:
+  datadir: "/tmp/shinesolutions/packer-aem/hieradata"
+  data_hash: yaml_data
+hierarchy:
+  - name: Hiera Data
+    paths: 
+      - aws.yaml
```
Comment (cliffano) [inline conf/puppet/hiera.yaml]: Why are aws.yaml, docker.yaml, and author-publish-dispatcher.yaml specified here? Shouldn't they be loaded from `%{::platform_type}.yaml` and `%{::component}.yaml` ?

Comment (phillipi-shinesolutions) [reply conf/puppet/hiera.yaml]: Puppet fails to find values without the additional hieras being specifically specified in global hiera

**Pull request merged**